### PR TITLE
topsql: introduce sql exec count and sql duration

### PR DESCRIPTION
**Pull request by mornyx**
Signed-off-by: mornyx <mornyx.z@gmail.com>

### What's changed

- Added `TopSQLRecord`, `CPUTimeRecord` has been removed.
- Added `TopSQLSubResponse.TopSQLRecord`, `TopSQLSubResponse.CPUTimeRecord` has been removed.
- Added `TopSQLAgent.ReportTopSQLRecords()`, `TopSQLAgent.ReportCPUTimeRecords()` has been removed.

### CODE REVIEW

**Comment by mornyx** [comment]
PTAL, Thanks~
/cc @crazycs520 @zhongzc @breeswish

File: proto/topsql_agent.proto [hunk 1]
```diff
@@ -55,8 +73,9 @@ message TopSQLSubRequest {}
 
 message TopSQLSubResponse {
     oneof resp_oneof {
-        CPUTimeRecord record = 1;
+        CPUTimeRecord record = 1; // DEPRECATED: use TopSQLRecord instead
```
Comment (breeswish) [inline proto/topsql_agent.proto:76]: Maybe we can drop it, as there is no existing user.

File: proto/topsql_agent.proto [hunk 2]
```diff
@@ -13,8 +13,11 @@ option (gogoproto.unmarshaler_all) = true;
 
 // TopSQLAgent is the persistent agent service for TopSQL records
 service TopSQLAgent {
+    // DEPRECATED: use ReportTopSQLRecords instead.
     // ReportCPUTimeRecords is called periodically (e.g. per minute) to save the in-memory TopSQL records
     rpc ReportCPUTimeRecords(stream CPUTimeRecord) returns (EmptyResponse) {}
```
Comment (breeswish) [inline proto/topsql_agent.proto:18]: Seems that we will never call this API more. If so, then maybe you can remove this interface.

**Review by breeswish — COMMENTED** [review]
the rest LGTM

**Comment by breeswish** [comment]
You also need to sync the data structure of https://github.com/pingcap/kvproto/blob/master/proto/resource_usage_agent.proto with this PR, as now this is our first public release API, and any inconformity will become a debt.

File: proto/topsql_agent.proto (hunk 1 above)
Comment (mornyx) [reply proto/topsql_agent.proto]: OK, I have deleted it and tested on TiDB and NGM.

File: proto/topsql_agent.proto (hunk 2 above)
Comment (mornyx) [reply proto/topsql_agent.proto]: OK, I have deleted it and tested on TiDB and NGM.

**Comment by mornyx** [comment]
> You also need to sync the data structure of https://github.com/pingcap/kvproto/blob/master/proto/resource_usage_agent.proto with this PR, as now this is our first public release API, and any inconformity will become a debt.

cc @zhongzc 

I suggest change to:
```protobuf
message ResourceUsageRecord {
     bytes resource_group_tag = 1;
     repeated ResourceUsageRecordItem items = 2;
}

message ResourceUsageRecordItem {
     uint64 timestamp_sec = 1;
     uint32 cpu_time_ms = 2;
     uint32 read_keys = 3;
     uint32 write_keys = 4;
}
```

to keep consistency with `tipb`.

**Review by breeswish — APPROVED** [review]

File: proto/topsql_agent.proto [hunk 3]
```diff
@@ -13,8 +13,8 @@ option (gogoproto.unmarshaler_all) = true;
 
 // TopSQLAgent is the persistent agent service for TopSQL records
 service TopSQLAgent {
-    // ReportCPUTimeRecords is called periodically (e.g. per minute) to save the in-memory TopSQL records
-    rpc ReportCPUTimeRecords(stream CPUTimeRecord) returns (EmptyResponse) {}
```
Comment (XuHuaiyu) [inline proto/topsql_agent.proto:17]: I think it's a bad practice that we rename the attribute directly. 😢
@breeswish

Comment (breeswish) [reply proto/topsql_agent.proto:17]: Yes I agree. This is a breaking change and should be performed with careful review.

**Pull request merged**